R/Ameriflux_data_tables.R

```diff
@@ -0,0 +1,562 @@
+      QUALIFIERS_EXIST <- TRUE
+    }
+    
+    else if (str_sub(att,-3,-1) %in% c('_PI','_QC','_IU','_SD')){
+      x <- str_sub(att,-3,-1)
+      extra <- definitions[which(definitions$uniqueAttributeLabel == x), 'uniqueAttributeDefinition']
+      str_sub(att,-3,-1) <- ""
+      QUALIFIERS_EXIST <- TRUE
+    }
+    
+    else if (str_sub(att,-2,-1) == '_F'){
+      x <- '_F'
+      extra <- definitions[which(definitions$uniqueAttributeLabel == '_F'), 'uniqueAttributeDefinition']
+      str_sub(att,-2,-1) <- ""
+      QUALIFIERS_EXIST <- TRUE
+    }
+    
+    else if (str_sub(att,-2,-1) %in% c('_1','_2','_3','_4','_5','_6','_7','_8','_9')){
+      x <- '_#'
+      extra <- definitions[which(definitions$uniqueAttributeLabel == '_#'), 'uniqueAttributeDefinition']
+      str_sub(att,-2,-1) <- ""
+      QUALIFIERS_EXIST <- TRUE
+    }
+    
+    else if (str_sub(att,-2,-1) %in% c('_R','_A')){
+      x <- paste('_H_V', str_sub(att,-2,-1), sep = '')
+      extra <- definitions[which(definitions$uniqueAttributeLabel == x), 'uniqueAttributeDefinition']
+      str_sub(att,-6,-1) <- ""
+      QUALIFIERS_EXIST <- TRUE
+    }
+    
+    else {
+      QUALIFIERS_EXIST <- FALSE
+    }
+    
+    #if the attribute (with potential qualifiers removed) 
+    #is not defined in definitions, there is a problem
+    
+    if (!(att %in% definitions$uniqueAttributeLabel)){
+      print(paste('Error: The attribute: ', 
+                  att, 
+                  ' is not defined in the definitions dataframe.',
+                  sep = ''))
+    }
+    
+    #if there were qualifiers, include the extra part of the definition
+    #when populating defs
+    else if (QUALIFIERS_EXIST){
+      #def is the corresponding definition for i-th attribute
+      def <- paste(definitions[which(definitions$uniqueAttributeLabel == att),'uniqueAttributeDefinition'],
+                   '; With qualifier \'',
+                   x,
+                   '\' ',
+                   extra,
+                   sep = '')
+      
+      #assign def to the corresponding position in defs
+      defs[i] <- def
+      i <- i+1
+    }
+    
+    #if there were no qualifiers, use regular definition
+    else {
+      def <- definitions[which(definitions$uniqueAttributeLabel == att),'uniqueAttributeDefinition']
+      #assign def to the corresponding position in defs
+      defs[i] <- def
+      i <- i+1
+    }
+    
+  }
+  return(defs)
+}
+
+##############################################################################
+
+
+
+
+
+
+
+
+
+
+##############################################################################
+
+#function definition for attribute_units
+#inputs:
+  #data: a dataframe with attributes to define
+  #definitions: dataframe in the style of definitions.csv
+#output: a vector of attribute definitions to be used in attribute table
+
+##############################################################################
+
+attribute_units <- function(data, definitions){
```

Comment:
It looks like this function searches for qualifiers in variable names and deletes them.  You could probably simplify this by using the gsub function and a regular expression.  Take a look at what this code does and you should be able to simplify this quite a bit.  | stands for "or" in R regular expressions.  
```
names <- c("var_1", "var_F_2")
gsub("_1|_2|_F", "", names)
```

https://www.rstudio.com/wp-content/uploads/2016/09/RegExCheatsheet.pdf